src/include/miopen/conv/heuristic_model/tuning_heuristic.hpp

```diff
@@ -0,0 +1,119 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2023 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#ifndef GAURD_MIOPEN_TUNING_HEURISTIC_HPP_
+#define GAURD_MIOPEN_TUNING_HEURISTIC_HPP_
+
+#if MIOPEN_ENABLE_AI_HEUR
+#include <fdeep/fdeep.hpp>
+#include <fdeep/tensor.hpp>
+#include <fdeep/tensor_shape.hpp>
+#include <miopen/conv/context.hpp>
+#include <miopen/solver.hpp>
+#include <unordered_map>
+#include <queue>
+#include <typeinfo>
+#include <string>
+
+namespace miopen {
+namespace ai {
+
+inline nlohmann::json get_metadata(const std::string& arch, const std::string& solver)
+{
+    std::string file_path = GetSystemDbPath() + "/" + arch + "_" + solver + "_metadata.model";
+    return nlohmann::json::parse(std::ifstream(file_path));
+}
+
+inline fdeep::model
+get_model(const std::string& arch, const std::string& solver, const std::string& model_type)
```

Comment:
1. Please use `CamelCase` for all function names (like in other places).
2. From the function name it is not clear what kind of model we are getting.
3. Do you need this function to be inline? If not please put only declaration in the header.

Reply:
Let's fix this later.

Reply:
@atamazov Why not do it in this PR?

@Dmantri98 What difficulties do you have? Do you need assistance?

Reply:
@averinevg Because the users need this asap. Please discuss this with @JehandadKhan

Reply:
@averinevg Because the users need #2055, but #2055 and this PR share a lot of the common code. Please discuss details with @JehandadKhan

Reply:
I will introduce a larger change that will get rid of the need for including tuning_heuristic.hpp within conv_asm_1x1u.cpp. This is the main reason for keeping the inline function since fdeep is a header only library and including it twice, which would happen if we split out the functions that need fdeep::model to be defined into tuning_heuristic.cpp, would cause a duplicate definition error as fdeep.hpp would need to be included in tuning_heuristic.cpp and conv_asm_1x1U.cpp.

Reply:
@Dmantri98 
> This is the main reason for keeping the inline function since fdeep is a header only library and including it twice... would cause a duplicate definition error

This reason is invalid. fdeep should be fixed in order to allow multiple inclusions of its public header into the sources.

Reply:
@Dmantri98 Pleae let me know if you need assistance with fdeep.

Reply:
Resolved.